lib/user_interface.ex

```diff
@@ -0,0 +1,34 @@
+defmodule UserInterface do
+  def get_input(message, io \\ IO) do
+    message
+    |> io.gets()
+    |> String.trim()
+    |> String.first()
+    |> get_input(message, io)
+  end
+
+  def get_input(nil, _, io) do
+    get_input(
+      "I didn't understand that and I don't know what to do. Please enter a valid command.",
```

Comment:
A little bit of duplicate code here.  What's the reasoning behind this inclusion?

Reply:
I don't have a good reason. Just updated it to send it to the same error-handling path all other invalid user inputs go to.